io/readers.ts

```diff
@@ -0,0 +1,36 @@
+import { Buffer, Closer, read, Reader, ReadResult, Writer } from "deno";
+
+const encoder = new TextEncoder();
+
+export class StringReader implements Reader {
+  private offs = 0;
```

Comment:
We should use the `_` sigil to preface private members.

Reply:
We doesn't have any convention about naming of private members. IMO, `_` is redundant if there are no special reason.

Reply:
Ya... we need to decide one way or another at some point on this. We've got code that does both. Obviously @kitsonk is on the side of `_` ... I don't particularly care as long as its uniform ... I think we need a style guide PR and the ensuing <s>blood bath</s> discussion to figure it out : )

Reply:
I have argued multiple times, there are two special reasons. To which no one has had an argument against.

- On TypeScript private members take up a slot on the API surface, because they aren't really private. So you could be squatting on something semantically meaningful in a subclass. While this is likely rare in practice, it is still good form.
- When private members do arrive, they will use the `#` sigil. To use them, we have to refactor at both the declaration and call site. Using `_` means easier refactoring down the road.

Plus there is that it has been a common convention for a long long time and it makes the intent at the call site more clear, that you are accessing a private member.

Reply:
Anyway, those discussion should not be concluded at here. Let's have discussion on another issue.